discovery/gossiper.go

```diff
@@ -2069,17 +2111,37 @@ func (d *AuthenticatedGossiper) updateChannel(info *channeldb.ChannelEdgeInfo,
 	if info.AuthProof != nil {
 		chanID := lnwire.NewShortChanIDFromInt(info.ChannelID)
 		chanAnn = &lnwire.ChannelAnnouncement{
-			NodeSig1:       info.AuthProof.NodeSig1,
-			NodeSig2:       info.AuthProof.NodeSig2,
 			ShortChannelID: chanID,
-			BitcoinSig1:    info.AuthProof.BitcoinSig1,
-			BitcoinSig2:    info.AuthProof.BitcoinSig2,
-			NodeID1:        info.NodeKey1,
-			NodeID2:        info.NodeKey2,
+			NodeID1:        info.NodeKey1Bytes,
+			NodeID2:        info.NodeKey2Bytes,
 			ChainHash:      info.ChainHash,
-			BitcoinKey1:    info.BitcoinKey1,
+			BitcoinKey1:    info.BitcoinKey1Bytes,
 			Features:       lnwire.NewRawFeatureVector(),
-			BitcoinKey2:    info.BitcoinKey2,
+			BitcoinKey2:    info.BitcoinKey2Bytes,
+		}
+		chanAnn.NodeSig1, err = lnwire.NewSigFromRawSignature(
```

Comment:
again, I would prefer doing this before creating the `ChannelAnnouncement`, even though it would create more temp variables.